src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = calculateBucketIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> prevNode = null;
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                prevNode = currentNode;
+                currentNode = currentNode.next;
+            }
+            prevNode.next = newNode;
+        }
+        size++;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getForNullKey();
+        }
+        int index = calculateBucketIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private V getForNullKey() {
+        if (table[0] != null) {
+            Node<K, V> currentNode = table[0];
+            while (currentNode != null) {
+                if (currentNode.key == null) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            }
+        }
         return null;
     }
 
+    private void resize() {
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
```

Comment:
```suggestion
        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
```
By doing so, we can do resize at the beginning of the put method. It is beneficial as we won't need to add an element, which cause the resize, twice